model/context.go

```diff
@@ -107,10 +107,37 @@ func truncate(s string) string {
 	return s
 }
 
+type NetworkInfo struct {
+	EffectiveType *string
+	RoundTripTime *int64
+	Downlink      *int64
+	DownlinkMax   *int64
+	SaveData      *bool
+	Type          *string
+}
+
+func (ni *NetworkInfo) fields() common.MapStr {
+	fields := common.MapStr{}
+	if ni == nil {
+		return fields
+	}
+	utility.Set(fields, "type", ni.Type)
+	utility.Set(fields, "downlink", ni.Downlink)
+	utility.Set(fields, "downlinkMax", ni.DownlinkMax)
+	utility.Set(fields, "effectiveType", ni.EffectiveType)
+	utility.Set(fields, "roundTripTime", ni.RoundTripTime)
+	utility.Set(fields, "saveData", ni.SaveData)
+	return fields
+}
+
 // Page consists of URL and referer
 type Page struct {
-	URL     *URL
-	Referer *string
+	URL                    *URL
+	Referer                *string
+	NetworkInfo            *NetworkInfo
+	Cores                  *int
+	Memory                 *int
+	ServedViaServiceWorker *string
```

Comment:
in https://github.com/elastic/apm-server/issues/3657#issuecomment-615128282 @vigneshshanmugam said we're still exploring whether this should be a boolean or keyword. Has that been clarified since?

Reply:
It will be string with predefined values. Updated the description with more context.